app/models/binary/DataSetService.scala

```diff
@@ -116,81 +116,92 @@ class DataSetService @Inject()(organizationDAO: OrganizationDAO,
   }
 
   def updateDataSources(dataStore: DataStore, dataSources: List[InboxDataSource])(
-      implicit ctx: DBAccessContext): Fox[List[Unit]] = {
+      implicit ctx: DBAccessContext): Fox[List[ObjectId]] = {
     logger.info(
       s"[${dataStore.name}] Available datasets: " +
         s"${dataSources.count(_.isUsable)} (usable), ${dataSources.count(!_.isUsable)} (unusable)")
 
     val groupedByOrga = dataSources.groupBy(_.id.team).toList
 
-    Fox.serialCombined(groupedByOrga) { orgaTuple: (String, List[InboxDataSource]) =>
-      for {
-        organizationBox <- organizationDAO.findOneByName(orgaTuple._1).futureBox
-        result <- organizationBox match {
-          case Full(organization) =>
-            for {
-              foundDatasets <- dataSetDAO.findAllByNamesAndOrganization(orgaTuple._2.map(_.id.name), organization._id)
-              foundDatasetsByName = foundDatasets.groupBy(_.name)
-              _ <- Fox.serialSequence(orgaTuple._2)(dataSource =>
-                updateDataSource(dataStore, dataSource, foundDatasetsByName))
```

Comment:
@fm3 I found the error because this `Fox.serialSequence` gets evaluated to a `Future[List[Box[Unit]]]`, so a `Failure` inside the boxes wouldn't be noticed.